feat: Organization VPCs and org VPC peering (#579)
aiven/aiven-docs: docs/platform/howto/manage-organization-vpc.md

- `NETWORK_CIDR` with the CIDR block (a range of IP addresses) for the VPC, for example,
  `10.0.0.0/24`
- `ORGANIZATION_ID` with the ID of your Aiven organization where to create the VPC, for
  example, `org1a2b3c4d5e6`

</TabItem>
-->
<TabItem value="api" label="Aiven API">

Make an API call to the `OrganizationVpcCreate` endpoint:

```bash
curl --request POST \
  --url https://api.aiven.io/v1/organization/ORGANIZATION_ID/vpcs \
  --header 'Authorization: Bearer BEARER_TOKEN' \
  --header 'content-type: application/json' \
  --data '
    {
      "cloud_name": "CLOUD_PROVIDER_REGION",
      "network_cidr": "NETWORK_CIDR"
    }
  '
```

Replace the following placeholders with meaningful data:

- `ORGANIZATION_ID`
- `BEARER_TOKEN`
- `CLOUD_PROVIDER_REGION`
- `NETWORK_CIDR`

</TabItem>
<TabItem value="tf" label="Aiven Provider for Terraform">
Use the `aiven_organization_vpc` resource. See the
[Aiven Provider for Terraform documentation](https://registry.terraform.io/providers/aiven/aiven/latest/docs)
for details.
</TabItem>
</Tabs>

## Create a service in an organization VPC

Your organization VPC is available as a geolocation (cloud region) for the new service.

:::note

> You can only create a service in an organization VPC if the organization VPC is located in
> the organization where to create the service.

Reviewer: ```suggestion
You can create a service in an organization VPC only if the organization VPC is in the 
same organization where you are creating the service.
```
For more clarity.

</TabItem>
-->
<TabItem value="api" label="Aiven API">

Make an API call to the
[ServiceCreate endpoint](https://api.aiven.io/doc/#tag/Service/operation/ServiceCreate)
endpoint:

```bash {12}
curl --request POST \
  --url https://api.aiven.io/v1/project/PROJECT_NAME/service \
  --header 'Authorization: Bearer BEARER_TOKEN' \
  --header 'content-type: application/json' \
  --data-raw '
    {
      "service_name": "SERVICE_NAME",
      "cloud": "CLOUD_PROVIDER_REGION",
      "plan": "SERVICE_PLAN",
      "service_type": "SERVICE_TYPE",
      "disk_space_mb": DISK_SIZE,
      "project_vpc_id":"ORGANIZATION_VPC_ID"
    }
  '
```

Replace the following placeholders with meaningful data:

- `PROJECT_NAME`, for example `org-vpc-test`
- `BEARER_TOKEN`
- `SERVICE_NAME`, for example `org-vpc-test-project`
- `CLOUD_PROVIDER_REGION`, for example `google-europe-west10`
- `SERVICE_PLAN`, for example `startup-4`
- `SERVICE_TYPE`, for example `pg`
- `DISK_SIZE` in MiB, for example `81920`
- `ORGANIZATION_VPC_ID`

</TabItem>
</Tabs>

## Migrate a service to an organization VPC

Your organization VPC is available as a geolocation (cloud region) for your service.

:::note

> You can only migrate a service to an organization VPC if the organization VPC is located in
> the organization where your service runs.

Reviewer: ```suggestion
You can only migrate a service to an organization VPC if the organization VPC is in the 
same organization where your service runs.
```

:::

Migrate a service to an organization VPC using a tool of your choice:

<Tabs groupId="group1">
<TabItem value="console" label="Aiven Console" default>

1. In [Aiven Console](https://console.aiven.io/), open your service and click
   <ConsoleLabel name="Service settings"/>.
1. In the **Cloud and network** section, click <ConsoleLabel name="actions"/> >
   **Change cloud or region**.
1. In the **Region** section, go to the **VPCs** tab, select your organization VPC and
   click **Migrate**.

</TabItem>
<!--
<TabItem value="cli" label="Aiven CLI">

Run [avn service update](/docs/tools/cli/service-cli#avn-cli-service-update):

```bash
avn service update SERVICE_NAME \

>   --organization-vpc-id ORGANIZATION_VPC_ID

Reviewer 2: In the service API here we still have `project-vpc-id` here and `project_vpc_id` in here, even if the value can be an ID of an organization ID too. This is for API compatibility as a rename in API would be a breaking change, so I elected to not do it.
